cmd/serviceusers/update_test.go

```diff
@@ -0,0 +1,197 @@
+			wantErr:    true,
+			wantErrOut: regexp.MustCompile(`no key or value provided for label`),
+			httpStubs: []httpmock.Stub{
+				{
+					URL: "/service-users/1767c791-4001-429b-82cf-0a471cc2f5d2",
+					Responder: func(w http.ResponseWriter, r *http.Request) {
+						if r.Method == http.MethodGet {
+							filename := "../../pkg/serviceusers/fixtures/service-user-get.json"
+							f, err := os.Open(filename)
+							if err != nil {
+								panic(fmt.Sprintf("Internal testing error: could not open %q", filename))
+							}
+							defer f.Close()
+							w.Header().Set("Content-Type", "application/json")
+							w.WriteHeader(http.StatusOK)
+							reader := bufio.NewReader(f)
+							content, err := io.ReadAll(reader)
+							if err != nil {
+								panic(fmt.Sprintf("Internal testing error: could not read %q", filename))
+							}
+							fmt.Fprint(w, string(content))
+							return
+						}
+					},
+				},
+			},
+		},
+		{
+			desc:       "invalid tag arguments",
+			args:       []string{"update", "1767c791-4001-429b-82cf-0a471cc2f5d2", "add", "tag"},
+			wantErr:    true,
+			wantErrOut: regexp.MustCompile(`not enough arguments`),
+			httpStubs: []httpmock.Stub{
+				{
+					URL: "/service-users/1767c791-4001-429b-82cf-0a471cc2f5d2",
+					Responder: func(w http.ResponseWriter, r *http.Request) {
+						if r.Method == http.MethodGet {
+							filename := "../../pkg/serviceusers/fixtures/service-user-get.json"
+							f, err := os.Open(filename)
+							if err != nil {
+								panic(fmt.Sprintf("Internal testing error: could not open %q", filename))
+							}
+							defer f.Close()
+							w.Header().Set("Content-Type", "application/json")
+							w.WriteHeader(http.StatusOK)
+							reader := bufio.NewReader(f)
+							content, err := io.ReadAll(reader)
+							if err != nil {
+								panic(fmt.Sprintf("Internal testing error: could not read %q", filename))
+							}
+							fmt.Fprint(w, string(content))
+							return
+						}
+					},
+				},
+			},
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			cmd, registry, stdout, teardown := setupServiceUsersTest(t, &tC)
+			defer func() {
+				registry.Teardown()
+				teardown()
+			}()
+
+			_, err := cmd.ExecuteC()
+			if (err != nil) != tC.wantErr {
+				t.Logf("Stdout: %s", stdout)
+				t.Fatalf("TestServiceUsersUpdate() error = %v, wantErr %v", err, tC.wantErr)
+			}
+			if (err != nil) == tC.wantErr && tC.wantErrOut != nil {
+				errString := err.Error()
+				matchString := tC.wantErrOut.String()
+				if !tC.wantErrOut.MatchString(errString) {
+					t.Fatalf("\n%s\n != \n%s", errString, matchString)
+				}
+			}
+			if tC.wantOut != nil {
+				parsedOut := stdout.String()
+				if !tC.wantOut.MatchString(parsedOut) {
+					t.Fatalf("\n%s\n != \n%s", tC.wantOut.String(), parsedOut)
+				}
+			}
+		})
+	}
+}
+
+func defaultUpdateResponseHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodGet {
```

Comment:
we can reduce the size signfiicantly here
for example

```golang
func defaultUpdateResponseHandler(w http.ResponseWriter, r *http.Request) {
	var filename string
	if r.Method == http.MethodGet {
		filename := "../../pkg/serviceusers/fixtures/service-user-get.json"
	}
	if r.Method == http.MethodPut {
		filename := "../../pkg/serviceusers/fixtures/service-user-update.json"
	}
	f, err := os.Open(filename)
	if err != nil {
		panic(fmt.Sprintf("Internal testing error: could not open %q", filename))
	}
	defer f.Close()
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)
	reader := bufio.NewReader(f)
	content, err := io.ReadAll(reader)
	if err != nil {
		panic(fmt.Sprintf("Internal testing error: could not read %q", filename))
	}
	fmt.Fprint(w, string(content))
}

```

Reply:
Makes a whole lot of sense